From 7108f83c8ad1bad4072e4f32da3db6d59cf51400 Mon Sep 17 00:00:00 2001
From: Ivan Murashov <ivan.murashov@lge.com>
Date: Tue, 20 Jul 2021 13:16:44 +0300
Subject: [PATCH] GCC: Remove double declaration of ContextSet

After the CL
https://chromium-review.googlesource.com/c/angle/angle/+/2965780
the build with GCC failed with error:
/third_party/angle/src/libANGLE/Display.h:325:37: error: declaration of
'typedef class std::__1::set<gl::Context*> egl::Display::ContextSet'
changes meaning of 'ContextSet' [-fpermissive]
/third_party/angle/src/libANGLE/Display.h:75:7: note: 'ContextSet'
declared here as 'using ContextSet = class std::__1::set<gl::Context*>'

To fix the error the double declaration of ContextSet is removed.

Bug: angleproject:5878, chromium:819294
Change-Id: Id9e52061af53ea18dd5d13b960daaa67a14f61ca
Reviewed-on: https://chromium-review.googlesource.com/c/angle/angle/+/3038804
Reviewed-by: Jamie Madill <jmadill@chromium.org>
Commit-Queue: Jamie Madill <jmadill@chromium.org>
---
 CONTRIBUTORS           | 1 +
 src/libANGLE/Display.h | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/CONTRIBUTORS b/CONTRIBUTORS
index 887ddc266..94b1b4d1d 100644
--- a/third_party/angle/CONTRIBUTORS
+++ b/third_party/angle/CONTRIBUTORS
@@ -154,6 +154,7 @@ Advanced Micro Devices, Inc.
 
 LG Electronics, Inc.
  Jani Hautakangas
+ Ivan Murashov
 
 IBM Inc.
  Junliang Yan
diff --git a/src/libANGLE/Display.h b/src/libANGLE/Display.h
index f33123b71..f0c091044 100644
--- a/third_party/angle/src/libANGLE/Display.h
+++ b/third_party/angle/src/libANGLE/Display.h
@@ -322,7 +322,6 @@ class Display final : public LabeledObject,
 
     ConfigSet mConfigSet;
 
-    typedef std::set<gl::Context *> ContextSet;
     ContextSet mContextSet;
 
     typedef std::set<Image *> ImageSet;
-- 
2.31.1

